OPAL/br/src/main/scala/org/opalj/br/fpcf/properties/fieldaccess/FieldAccessInformation.scala

```diff
@@ -0,0 +1,114 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj
+package br
+package fpcf
+package properties
+package fieldaccess
+
+import org.opalj.br.DefinedMethod
+import org.opalj.br.Field
+import org.opalj.br.PC
+import org.opalj.br.PCs
+import org.opalj.br.analyses.DeclaredMethods
+import org.opalj.collection.immutable.LongLinkedSet
+import org.opalj.collection.immutable.LongLinkedTrieSet
+import org.opalj.fpcf.Entity
+import org.opalj.fpcf.FallbackReason
+import org.opalj.fpcf.OrderedProperty
+import org.opalj.fpcf.PropertyIsNotDerivedByPreviouslyExecutedAnalysis
+import org.opalj.fpcf.PropertyKey
+import org.opalj.fpcf.PropertyMetaInformation
+import org.opalj.fpcf.PropertyStore
+
+sealed trait FieldAccessInformationPropertyMetaInformation extends PropertyMetaInformation {
+
+    final type Self = FieldAccessInformation
+}
+
+/**
+ * Describes all read and write accesses to a [[org.opalj.br.Field]].
+ *
+ * @author Maximilian Rüsch
+ */
+case class FieldAccessInformation(
+        private val encodedReadAccesses:  LongLinkedSet,
+        private val encodedWriteAccesses: LongLinkedSet
+) extends OrderedProperty
+    with FieldAccessInformationPropertyMetaInformation {
+
+    override def checkIsEqualOrBetterThan(e: Entity, other: FieldAccessInformation): Unit = {
+        if (numReadAccesses > other.numReadAccesses || numWriteAccesses > other.numWriteAccesses) {
+            throw new IllegalArgumentException(s"$e: illegal refinement of $other to $this")
+        }
+    }
+
+    final def key: PropertyKey[FieldAccessInformation] = FieldAccessInformation.key
+
+    def equals(other: FieldAccessInformation): Boolean =
```

Comment:
I don't think this is necessary, it should be exactly what the case class provides already.